current/runtime/storage.md

```diff
@@ -4,49 +4,237 @@ lang: en
+[the advanced storage documentation](../advanced/storage) to learn more about how Substrate's
+Storage Maps are implemented.
+
+#### Methods
+
+[Storage Maps expose an API](https://substrate.dev/rustdocs/master/frame_support/storage/trait.StorageMap.html#required-methods)
+that is similar to that of Storage Values.
+
+- `get` - Load the value associated with the provided key from storage. Docs:
+  [`StorageMap#get(key)`](https://substrate.dev/rustdocs/master/frame_support/storage/trait.StorageMap.html#tymethod.get),
+  [`StorageDoubleMap#get(key1, key2)`](https://substrate.dev/rustdocs/master/frame_support/storage/trait.StorageDoubleMap.html#tymethod.get)
+- `insert` - Store the provided value by associating it with the given key. Docs:
+  [`StorageMap#insert(key, val)`](https://substrate.dev/rustdocs/master/frame_support/storage/trait.StorageMap.html#tymethod.insert),
+  [`StorageDoubleMap#insert(key1, key2, val)`](https://substrate.dev/rustdocs/master/frame_support/storage/trait.StorageDoubleMap.html#tymethod.insert)
+- `mutate` - Use the provided function to mutate the value associated with the given key. Docs:
+  [`StorageMap#mutate(key, fn)`](https://substrate.dev/rustdocs/master/frame_support/storage/trait.StorageMap.html#tymethod.mutate),
+  [`StorageDoubleMap#mutate(key1, key2, fn)`](https://substrate.dev/rustdocs/master/frame_support/storage/trait.StorageDoubleMap.html#tymethod.mutate)
+- `take` - Load the value associated with the given key and remove it from storage. Docs:
+  [`StorageMap#take(key)`](https://substrate.dev/rustdocs/master/frame_support/storage/trait.StorageMap.html#tymethod.take),
+  [`StorageDoubleMap#take(key1, key2)`](https://substrate.dev/rustdocs/master/frame_support/storage/trait.StorageDoubleMap.html#tymethod.take)
+
+#### Iterable Storage Maps
+
+Depending on [the hashing algorithm](#Transparent-Hashing-Algorithms) that you select to generate a
+map's keys, you may be able to iterate across its keys and values. Because maps are often used to
+track unbounded sets of data (account balances, for example) it is especially likely to exceed block
+production time by iterating over maps in their entirety within the runtime. Furthermore, because
+accessing the elements of a map requires more pointer dereferencing than accessing the elements of a
+native list, maps are significantly _more_ costly than lists to iterate over with respect to time.
+This is not to say that it is "wrong" to iterate over maps in your runtime; in general Substrate
+focuses on "[first principles](#Best-Practices)" as opposed to hard and fast rules of right and
+wrong. Being efficient within the runtime of a blockchain is an important first principle of
+Substrate and this information is designed to help you understand _all_ of Substrate's storage
+capabilities and use them in a way that respects the important first principles around which they
+were designed.
```

Comment:
would be nice to mention here when and which maps are iterable at the end of the day.

Reply:
This is a great point and I'm starting to wonder if we should even document non-transparent hashers since they are now deprecated. Should we just behave as if all hashers are transparent and all maps are iterable?

Reply:
Asked @shawntabrizi about this offline and he agrees that we should take the simpler approach: only document that which is not deprecated :+1:

Reply:
Tried to clarify this in the docs :+1: